dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java

```diff
@@ -435,19 +435,25 @@ public String getRepositoryId(Entry entry) {
 
     @Override
     public String getMainBranch(Entry entry, String repositoryId) {
-        RepositoriesApi api = new RepositoriesApi(apiClient);
         // Is default version set?
         if (entry.getDefaultVersion() != null) {
             return getBranchNameFromDefaultVersion(entry);
         } else {
             // If default version is not set, need to find the main branch
-            try {
-                Repository repository = api.repositoriesUsernameRepoSlugGet(repositoryId.split("/")[0], repositoryId.split("/")[1]);
-                return repository.getMainbranch().getName();
-            } catch (ApiException e) {
-                LOG.error("Unable to retrieve default branch for repository " + repositoryId, e);
-                return null;
-            }
+            return getDefaultBranch(repositoryId);
+        }
+    }
+
+    @Override
+    public String getDefaultBranch(String repositoryId) {
+        RepositoriesApi api = new RepositoriesApi(apiClient);
+        Repository repository = null;
+        try {
+            repository = api.repositoriesUsernameRepoSlugGet(repositoryId.split("/")[0], repositoryId.split("/")[1]);
+            return repository.getMainbranch().getName();
```

Comment:
If there is no main branch, does the previous line always thrown an exception? Based line 455, it looks like it, but flagging because otherwise you risk an NPE here on the `.getName()`.

Reply:
Not sure, added null check